projects/storefrontlib/shared/components/media/media.service.ts

```diff
@@ -53,6 +76,67 @@ export class MediaService {
       return;
     }
 
+    const commonMediaProperties = this.getCommonMediaObject(
+      mediaContainer,
+      format,
+      alt,
+      role
+    );
+
+    if (!commonMediaProperties) {
+      return;
+    }
+
+    return {
+      ...commonMediaProperties,
+      srcset: this.resolveSrcSet(mediaContainer, format),
+    };
+  }
+
+  /**
+   * Returns a `Media` object with the main media (`src`) and various sources
+   * for specific formats for HTML `<picture>` element.
+   */
+  getMediaForPictureElement(
+    mediaContainer?: MediaContainer | Image,
+    format?: string,
+    alt?: string,
+    role?: string
+  ): Media | undefined {
+    if (!mediaContainer) {
+      return;
+    }
+
+    const commonMediaProperties = this.getCommonMediaObject(
+      mediaContainer,
+      format,
+      alt,
+      role
+    );
+
+    if (!commonMediaProperties) {
+      return;
+    }
+
+    return {
+      ...commonMediaProperties,
+      sources: this.resolveSources(mediaContainer, format),
+    };
+  }
+
+  /**
+   * Generates attributes common for `<img>` ang `<picture>`.
+   */
+  protected getCommonMediaObject(
+    mediaContainer?: MediaContainer | Image,
+    format?: string,
+    alt?: string,
+    role?: string
+  ): Media | undefined {
+    if (!mediaContainer) {
+      return;
+    }
```

Comment:
```suggestion
  protected getCommonMediaObject(
    mediaContainer: MediaContainer | Image,
    format?: string,
    alt?: string,
    role?: string
  ): Media {

    const mainMedia: Image = mediaContainer.url
      ? mediaContainer
      : this.resolveMedia(mediaContainer as MediaContainer, format);

    return {
      src: this.resolveAbsoluteUrl(mainMedia?.url ?? ''),
      alt: alt ?? mainMedia?.altText,
      role: role ?? mainMedia?.role,
    };
  }
```
No need to check again if `mediaContainer` is defined as it was verified in `getMedia` and `getMediaForPictureElement`

Reply:
Removed